src/acl/agent.test.ts

```diff
@@ -848,6 +851,540 @@ describe("getAgentResourceAccessModesAll", () => {
+      "http://www.w3.org/ns/auth/acl#mode"
+    );
+    expect(addedQuads[1].object.value).toBe(
+      "http://www.w3.org/ns/auth/acl#Read"
+    );
+    expect(addedQuads[2].predicate.value).toBe(
+      "http://www.w3.org/ns/auth/acl#accessTo"
+    );
+    expect(addedQuads[2].object.value).toBe("https://arbitrary.pod/resource");
+    expect(addedQuads[3].predicate.value).toBe(
+      "http://www.w3.org/ns/auth/acl#agent"
+    );
+    expect(addedQuads[3].object.value).toBe(
+      "https://some.pod/profileDoc#webId"
+    );
+  });
+
+  it("does not forget to add a Quad for Append access if Write access is not given", () => {
+    // This test is basically there to test for regressions
+    // if we ever try to be clever about inferring Append access
+    // (but we should be able to leave that to the server).
+    const sourceDataset = Object.assign(
+      getMockDataset("https://arbitrary.pod/resource.acl"),
+      { accessTo: "https://arbitrary.pod/resource" }
+    );
+
+    const updatedDataset = unstable_setAgentResourceAccessModes(
+      sourceDataset,
+      "https://some.pod/profileDoc#webId",
+      {
+        read: false,
+        append: true,
+        write: false,
+        control: false,
+      }
+    );
+
+    const updatedQuads: Quad[] = Array.from(updatedDataset);
+    expect(updatedQuads).toHaveLength(4);
+    expect(updatedQuads[0].predicate.value).toBe(
+      "http://www.w3.org/1999/02/22-rdf-syntax-ns#type"
+    );
+    expect(updatedQuads[0].object.value).toBe(
+      "http://www.w3.org/ns/auth/acl#Authorization"
+    );
+    expect(updatedQuads[1].predicate.value).toBe(
+      "http://www.w3.org/ns/auth/acl#mode"
+    );
+    expect(updatedQuads[1].object.value).toBe(
+      "http://www.w3.org/ns/auth/acl#Append"
+    );
+    expect(updatedQuads[2].predicate.value).toBe(
+      "http://www.w3.org/ns/auth/acl#accessTo"
+    );
+    expect(updatedQuads[2].object.value).toBe("https://arbitrary.pod/resource");
+    expect(updatedQuads[3].predicate.value).toBe(
+      "http://www.w3.org/ns/auth/acl#agent"
+    );
+    expect(updatedQuads[3].object.value).toBe(
+      "https://some.pod/profileDoc#webId"
+    );
+  });
+
+  it("replaces existing Quads defining Access Modes for this agent", () => {
+    const sourceDataset = addAclRuleQuads(
+      getMockDataset("https://arbitrary.pod/resource.acl"),
+      "https://some.pod/profileDoc#webId",
+      "https://arbitrary.pod/resource",
+      { read: false, append: false, write: false, control: true },
+      "resource"
+    );
+
+    const updatedDataset = unstable_setAgentResourceAccessModes(
+      sourceDataset,
+      "https://some.pod/profileDoc#webId",
+      {
+        read: true,
+        append: false,
+        write: false,
+        control: false,
+      }
+    );
+
+    const updatedQuads: Quad[] = Array.from(updatedDataset);
+    expect(updatedQuads).toHaveLength(4);
+    expect(updatedQuads[0].predicate.value).toBe(
```

Comment:
Yep, fair point. In fact, I had a couple of fairly annoying sessions of re-ordering these, because doing TDD, I had no idea in what order they were going to be when writing the tests :P 

The disadvantage of using our own functions is that we're making these tests dependent on their functioning correctly as well, meaning that it will be harder to track down the source of an error when unit tests suddenly start failing en masse.

(And an alternative option would be to run `.find()` on `updatedQuads`, but then our test code is getting more complex and hence more likely to be buggy itself.)

The disadvantage of being order-dependent weighs about as strongly for me as the disadvantage of tests becoming less isolated, so which one I went with was a coin flip, really :P 

So... Not sure if there's a generic rule we can deduce here to align on as a team, but happy to follow whatever people decide.

Reply:
This is out of the scope of the PR anyways, but I'll bring it up as a discussion item to the team to get their feeling.